recipe_scrapers/_utils.py

```diff
@@ -65,64 +74,89 @@
 
 def _extract_fractional(input_string: str) -> float:
     input_string = input_string.strip()
-    if "/" in input_string:
-        # for example "1 1/2" is matched
-        components = input_string.split(" ")
-        whole_part, fractional_part = components[0], components[-1]
+
+    # Handling mixed numbers with unicode fractions e.g., '1⅔'
+    for unicode_fraction, fraction_part in FRACTIONS.items():
+        if unicode_fraction in input_string:
+            whole_number_part, _, _ = input_string.partition(unicode_fraction)
+
+            whole_number = float(whole_number_part or 0)
+            return whole_number + fraction_part
+
+    if input_string in FRACTIONS:
+        return FRACTIONS[input_string]
+
+    try:
+        return float(input_string)
+    except ValueError:
+        pass
+
+    if " " in input_string and "/" in input_string:
+        whole_part, fractional_part = input_string.split(" ", 1)
         numerator, denominator = fractional_part.split("/")
-        return float(whole_part) + float(int(numerator) / int(denominator))
+        return float(whole_part) + float(numerator) / float(denominator)
 
-    whole_part, fractional_amount = "", 0.0
-    for symbol in input_string:
-        if symbol in FRACTIONS:
-            fractional_amount += FRACTIONS[symbol]
-        else:
-            whole_part += symbol
+    elif "/" in input_string:
+        numerator, denominator = input_string.split("/")
+        return float(numerator) / float(denominator)
 
-    return float(whole_part) + float(fractional_amount)
+    raise ValueError(f"Unrecognized fraction format: '{input_string}'")
 
 
 def get_minutes(element):
     if element is None:
         raise ElementNotFoundInHtml(element)
 
-    # handle integer in string literal
+    # If element is a BeautifulSoup Tag, extract its text content
+    if hasattr(element, "text"):
+        element = element.text
+
     try:
         return int(element)
-    except Exception:
+    except ValueError:
         pass
 
-    if isinstance(element, str):
-        time_text = element
-    else:
-        time_text = element.get_text()
+    if not isinstance(element, str):
+        raise ValueError("Unexpected format for time element")
+    time_text = element
 
     # attempt iso8601 duration parsing
+    if "-" in time_text:  # sometimes formats are like this: '12-15 minutes'
+        _min, _, time_text = time_text.partition("-")
+    if " to " in time_text:  # sometimes formats are like this: '12 to 15 minutes'
+        _min, _to, time_text = time_text.partition(" to ")
+    if "PT0M" in time_text:
+        time_text = None
```

Comment:
As an (unlikely) counter-example, recipes _could_ theoretically list recipe timings using zero minutes plus seconds.  `PT0M120S` for example, for two minutes.

Even if that is unlikely, this early-return can probably be removed I think, unless it's to handle some particular case(s)?

Reply:
Was a temp fix for something fixed elsewhere